dell/src/main/java/org/apache/iceberg/dell/ecs/EcsCatalog.java

```diff
@@ -0,0 +1,520 @@
+        key.length() - TABLE_OBJECT_SUFFIX.length());
+    return TableIdentifier.of(namespace, tableName);
+  }
+
+  /**
+   * Remove table object. If the purge flag is set, remove all data objects.
+   */
+  @Override
+  public boolean dropTable(TableIdentifier identifier, boolean purge) {
+    if (!tableExists(identifier)) {
+      throw new NoSuchTableException("Table %s does not exist", identifier);
+    }
+
+    EcsURI tableObjectURI = tableURI(identifier);
+    if (purge) {
+      // if re-use the same instance, current() will throw exception.
+      TableOperations ops = newTableOps(identifier);
+      TableMetadata current = ops.current();
+      if (current == null) {
+        return false;
+      }
+
+      CatalogUtil.dropTableData(ops.io(), current);
+    }
+
+    client.deleteObject(tableObjectURI.bucket(), tableObjectURI.name());
+    return true;
+  }
+
+  private EcsURI tableURI(TableIdentifier id) {
+    EcsURI prefix = namespacePrefix(id.namespace());
+    // The prefix has the delimiter at the tail.
+    return new EcsURI(prefix.bucket(), prefix.name() + id.name() + TABLE_OBJECT_SUFFIX);
+  }
+
+  /**
+   * Table rename will only move table object, the data objects will still be in-place.
+   *
+   * @param from identifier of the table to rename
+   * @param to   new table name
+   */
+  @Override
+  public void renameTable(TableIdentifier from, TableIdentifier to) {
+    if (!namespaceExists(to.namespace())) {
+      throw new NoSuchNamespaceException("Cannot rename %s to %s because namespace %s does not exist",
+              from, to, to.namespace());
+    }
+
+    if (tableExists(to)) {
+      throw new AlreadyExistsException("Cannot rename %s because destination table %s exists", from, to);
+    }
+
+    EcsURI fromURI = tableURI(from);
+    if (!objectMetadata(fromURI).isPresent()) {
+      throw new NoSuchTableException("Cannot rename table because table %s does not exist", from);
+    }
+
+    Properties properties = loadProperties(fromURI);
+    EcsURI toURI = tableURI(to);
+
+    if (!putNewProperties(toURI, properties.content())) {
+      throw new AlreadyExistsException("Cannot rename %s because destination table %s exists", from, to);
+    }
+
+    client.deleteObject(fromURI.bucket(), fromURI.name());
+    LOG.info("rename table {} to {}", from, to);
+  }
+
+  @Override
+  public void createNamespace(Namespace namespace, Map<String, String> properties) {
+    EcsURI namespaceObject = namespaceURI(namespace);
+    if (!putNewProperties(namespaceObject, properties)) {
+      throw new AlreadyExistsException("namespace %s(%s) has already existed", namespace, namespaceObject);
+    }
+  }
+
+  private EcsURI namespaceURI(Namespace namespace) {
+    return new EcsURI(
+        dellProperties.ecsCatalogPrefix().bucket(),
+        dellProperties.ecsCatalogPrefix().name() +
+            String.join(dellProperties.ecsCatalogDelimiter(), namespace.levels()) +
+            NAMESPACE_OBJECT_SUFFIX);
```

Comment:
Nit:  Please use the `String.format` to concat the strings, that makes easier to read the pattern.